service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/NotifyGatekeeperEventHandler.java

```diff
@@ -6,33 +6,38 @@
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.fpl.events.NotifyGatekeepersEvent;
 import uk.gov.hmcts.reform.fpl.model.CaseData;
-import uk.gov.hmcts.reform.fpl.model.notify.SharedNotifyTemplate;
 import uk.gov.hmcts.reform.fpl.model.notify.sendtogatekeeper.NotifyGatekeeperTemplate;
+import uk.gov.hmcts.reform.fpl.service.FeatureToggleService;
 import uk.gov.hmcts.reform.fpl.service.email.NotificationService;
 import uk.gov.hmcts.reform.fpl.service.email.content.GatekeeperEmailContentProvider;
 
 import java.util.List;
 
 import static uk.gov.hmcts.reform.fpl.NotifyTemplates.GATEKEEPER_SUBMISSION_TEMPLATE;
+import static uk.gov.hmcts.reform.fpl.NotifyTemplates.GATEKEEPER_SUBMISSION_TEMPLATE_CHILD_NAME;
 import static uk.gov.hmcts.reform.fpl.utils.EmailNotificationHelper.getDistinctGatekeeperEmails;
 
 @Service
-@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
 public class NotifyGatekeeperEventHandler {
     private final NotificationService notificationService;
-    private final GatekeeperEmailContentProvider gatekeeperEmailContentProvider;
+    private final GatekeeperEmailContentProvider contentProvider;
+    private final FeatureToggleService toggleService;
 
     @EventListener
     public void notifyGatekeeper(NotifyGatekeepersEvent event) {
         CaseData caseData = event.getCaseData();
 
-        NotifyGatekeeperTemplate parameters = gatekeeperEmailContentProvider.buildGatekeeperNotification(caseData);
+        NotifyGatekeeperTemplate parameters = contentProvider.buildGatekeeperNotification(caseData);
 
         List<String> emailList = getDistinctGatekeeperEmails(caseData.getGatekeeperEmails());
 
+        String template = toggleService.isEldestChildLastNameEnabled() ? GATEKEEPER_SUBMISSION_TEMPLATE_CHILD_NAME
+                                                                       : GATEKEEPER_SUBMISSION_TEMPLATE;
+
         emailList.forEach(recipientEmail -> {
-            SharedNotifyTemplate template = parameters.duplicate();
-            notificationService.sendEmail(GATEKEEPER_SUBMISSION_TEMPLATE, recipientEmail, template, caseData.getId());
+            NotifyGatekeeperTemplate notifyData = parameters.duplicate();
```

Comment:
template?

Reply:
if by that you mean the var name should be `template` as it was initially then I have used that var name else where. I dislike that the notify data classes end in tempalte as they are not pointing to the template themselves, maybe a better name for the would be something along the lines of `NotifyGatekeeperTemplateData`. Will update the var name to reflect that suggested class name (i.e. `templateData`)